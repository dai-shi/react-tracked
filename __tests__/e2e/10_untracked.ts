/* global page */

jest.setTimeout(15 * 1000);

describe('10_untracked', () => {
  const port = process.env.PORT || '8080';

  it('should work with recorded events', async () => {
    await page.goto(`http://localhost:${port}/`);

    await page.waitForSelector('body > #app > ul > li:nth-child(2) > input:nth-child(2)');
    await page.click('body > #app > ul > li:nth-child(2) > input:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > ul > li:nth-child(2) > input:nth-child(2)', '2');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(3) > input:nth-child(2)');
    await page.click('body > #app > ul > li:nth-child(3) > input:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > ul > li:nth-child(3) > input:nth-child(2)', '3');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(2) > input:nth-child(1)');
    await page.click('body > #app > ul > li:nth-child(2) > input:nth-child(1)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(3) > input:nth-child(1)');
    await page.click('body > #app > ul > li:nth-child(3) > input:nth-child(1)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(1) > input:nth-child(1)');
    await page.click('body > #app > ul > li:nth-child(1) > input:nth-child(1)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(2) > button:nth-child(4)');
    await page.click('body > #app > ul > li:nth-child(2) > button:nth-child(4)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(3) > button:nth-child(3)');
    await page.click('body > #app > ul > li:nth-child(3) > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(3) > input:nth-child(1)');
    await page.click('body > #app > ul > li:nth-child(3) > input:nth-child(1)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > ul > li:nth-child(2) > button:nth-child(3)');
    await page.click('body > #app > ul > li:nth-child(2) > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();
  });
});
