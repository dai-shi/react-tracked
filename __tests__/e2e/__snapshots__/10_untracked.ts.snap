// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`10_untracked should work with recorded events 1`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Study JS\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Buy ticket\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`10_untracked should work with recorded events 2`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Buy ticket\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`10_untracked should work with recorded events 3`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Buy ticket\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`10_untracked should work with recorded events 4`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Buy ticket3\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`10_untracked should work with recorded events 5`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Buy ticket3\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`10_untracked should work with recorded events 6`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Buy ticket3\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`10_untracked should work with recorded events 7`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Buy ticket3\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`10_untracked should work with recorded events 8`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Buy ticket3\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`10_untracked should work with recorded events 9`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`10_untracked should work with recorded events 10`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`10_untracked should work with recorded events 11`] = `
"
    <div id=\\"app\\"><ul><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Wash dishes\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li><li><input type=\\"checkbox\\" checked=\\"\\"><input value=\\"Study JS2\\"><button type=\\"button\\">Delete</button><button type=\\"button\\">Duplicate</button></li></ul></div>
  <script src=\\"/main.js\\"></script>

"
`;
